Tools/PrivateRepoLib.ps1

```diff
@@ -0,0 +1,229 @@
+        
+        ## If a module is determined to be missing, throw an error
+        If($Result)
+        {
+            ## Modules have been identified as missing
+            $ErrorMessage = "`nMissing required PowerShell modules`n"
+            $ErrorMessage += "    Run the following command to install the missing modules: Install-Module Az`n`n"
+            
+            Write-Host $ErrorMessage -ForegroundColor Yellow
+            Throw "Unable to run script, missing required PowerShell modules"
+        }
+
+        ## Identifies if PowerShell session is currently connected to Azure.
+        $Result = Get-AzContext
+
+        If($null -eq $Result)
+        {
+            ## No active connections to Azure
+            Write-Host "Not connected to Azure, please connect to your Azure Subscription"
+            
+            ## Determines that a connection to Azure is neccessary, and if a Subscription Name was provided, connect to that Subscription
+            If($AzureSubscriptionName -eq "")
+                { Connect-AzAccount }
+            else 
+                { Connect-AzAccount -SubscriptionName $AzureSubscriptionName }
+
+            ## If the connection fails, or the user cancels the login request, then throw an error.
+            $Result = Get-AzContext
+            If($null -eq $Result)
+                { Throw "Failed to connect to Azure. Please run Connect-AzAccount to connect to Azure, or re-run the cmdlet and enter your credentials." }
+        }
+
+        ## Determines the PowerShell Parameter Set that was used in the call of this Function.
+        switch ($PsCmdlet.ParameterSetName) {
+            "WinGet" { 
+                ## Sets variables as if the Windows Package Manager was Private Repo Name was specified.
+                $AzureFunctionName      = $(Winget source list -n $PrivateRepoName)[4].split("/")[2].Split(".")[0]
+                $AzureResourceGroupName = $(Get-AzFunctionApp).Where({$_.Name -eq $AzureFunctionName}).ResourceGroupName
+             }
+            "Azure"  { 
+                ## Sets variables as if the Azure Function Name was provided.
+                $AzureResourceGroupName = $(Get-AzFunctionApp).Where({$_.Name -eq $AzureFunctionName}).ResourceGroupName
+             }
+        }
+
+        Write-Host "`nAzure Resources:"
+        Write-Host "  Azure Function Name: $AzureFunctionName"
+        Write-Host "  Azure Resource Group Name: $AzureResourceGroupName"
+
+        $TriggerName      = "ManifestPost"
+        $apiContentType   = "application/json"
+        $apiMethod        = "Post"
+
+        ## Creates the API Post Header
+        $apiHeader = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
+        $apiHeader.Add("Accept", 'application/json')
+
+        ## Gets the JSON Content for posting to Private Repo
+        $ApplicationManifest = Get-Content -Path $ManifestFilePath -Raw
+        $ApplicationManifest = $($ApplicationManifest -replace "`t|`n|`r|  ","")
```

Comment:
why is this necessary?

Reply:
When Invoking the rest method, errors were encountered when there were spaces or new lines present. The cmdlet appears to require a single line entry as the body of the rest method.